### ISSUE TITLE
Mixed octree and nodes

### ISSUE THREAD

**Issue by Pawlost** [starandserpent/Voxel-demo#83]
I think in future it may help with chunk manipulation
And by abstract I mean just use interface instead current node
And create mixed node which extends godot node and implements that interface
This means creating mixed replacable node
Its advantage would be making search for godot meshinstances more efficient
Because right now we have to do 2 searches
1. In our engine with octree complexity
2. In godot engine
Which has I think O(n) complexity
This would merge into one search

**Comment by Pawlost** [comment]
This is doable even with keeping it abstract, all godot related things are keeped here, if we made abstract terra nodes we could them then specify here and turn on godot every time we need to create new node. 
I think performance would be faster because when we hand stuff to godot, they stay there and we have to remove them (this is the reason I created this ticket). This means we have to remove it twice,  first in our code, second in godot. If we could remove things only once it would increase performance. Nodes should not take too much memory, but I am not sure about memory consuption.

**Comment by Pawlost** [comment]
This issue starts to be sensible when we start removing chunks. Right now it is not important.

**Comment by Pawlost** [comment]
Since we are using Visual Server which returns RID accessing godot Scene is not needed anymore. One ofmany  possible implementations is here https://github.com/starandserpent/Voxel-demo/issues/102 . For this reason I am closing this issue.
